### ISSUE TITLE
In overview mode, prevent keyboard input into windows

### ISSUE THREAD

**Issue by Acumane** [dawsers/hyprscroller#94]
Something like `plugin:scroller:overview_ignore_input` that prevents a focused window in the overview from receiving input from the keyboard. This is often the expected behavior of an overview like this.

Hyprland's `follow_mouse` distinguishes between mouse and keyboard focus, though I haven't checked whether this is exposed by the API.

**Comment by dawsers** [comment]
The main differentiating factor between *hyprscroller's* overview and others, like you mention, is that you can interact with the windows. To me, that is a feature instead of something bad. You can move, re-adapt, navigate the overview, or even close/add windows. If I disable keyboard input, it would only be a nice, useless overview of your workspace. Remember, most people use the keyboard instead of the mouse in a tiling window manager. I am even working on more features in that sense (interactivity and scaling). There is also `jump` if you just want to see the windows.

Overview has to hack mouse positioning and which monitor the cursor belongs to. To Hyprland, each monitor covers a disjoint area of the frame buffer, and both workspaces in a two monitor setup share frame buffer coordinate space. Instead, *hyprscroller* uses a virtual coordinate space that can be "infinite", and that doesn't map to any facility in Hyprland. The truth is Hyprland's way of doing things is a bit strange to me in the sense that it separates workspaces, but then shares the framebuffer for visible workspaces but not for unseen ones, though.

Basically, what I am trying to say is `overview` and `jump` modes use a very fragile mechanism that hacks and modifies Hyprland's code from the plugin. This can break at any point. Any modification to Hyprland can, and at some point will break *hyprscroller*, and I will have to find another way to make it work. It was already quite complicated to make it work relatively well, so I am not going to add even more "hacking into Hyprland" complexity to it., because it is a recipe for an even earlier disaster.

**Comment by Acumane** [comment]
I'm not asking for the keyboard to be disabled but rather for input not to be sent to the window from Hyprland. That's is, you could still resize or move focus, but not type something into the active window.

I mention `follow_mouse` because it suggests Hyprland makes this distinction already. Whether it's exposed by the API is another story.

**Comment by dawsers** [comment]
Yes, but unfortunately I cannot control what gets sent to the active window. If the keyboard is on, it just works, so it will either translate a key binding to move/change focus, or send the input to the active window if there is no key binding associated to what the user typed.
